shell/browser/api/atom_api_web_request_ns.cc

```diff
@@ -395,7 +418,11 @@ void WebRequestNS::OnListenerResult(uint64_t id,
       ReadFromResponse(isolate, &dict, out);
   }
 
-  std::move(callbacks_[id]).Run(result);
+  // The ProxyingURLLoaderFactory expects the callback to be executed
+  // asynchronously, because it used to work on IO thread before NetworkService.
+  base::SequencedTaskRunnerHandle::Get()->PostTask(
```

Comment:
What happens when its not executed async ?

Reply:
ProxyingURLLoaderFactory's logic is:
Post callback in next tick => pause network request => continue network request when callback is called

If we execute the callback in current tick then it becomes:
Run callback in current tick => continue network request(no-op) => pause network request

So the network request would be paused forever.

Reply:
Thanks for the explanation!